traffic_ops/traffic_ops_golang/cdni/shared.go

```diff
@@ -517,8 +539,20 @@ func checkBearerToken(bearerToken string, inf *api.APIInfo) (string, error) {
 	if dcdn != inf.Config.Cdni.DCdnId {
 		return "", errors.New("invalid token - incorrect dcdn")
 	}
+
+	if ucdn != inf.User.UCDN {
+		return "", errors.New("user ucdn did not match token ucdn")
+	}
+
 	if ucdn == "" {
-		return "", errors.New("invalid token - empty ucdn field")
+		if inf.User.Can("ICDN:UCDN-OVERRIDE") {
+			ucdn = inf.Params["ucdn"]
+			if ucdn == "" {
+				return "", errors.New("admin level ucdn requests require a ucdn query parameter")
+			}
+		} else {
+			return "", errors.New("invalid token - empty ucdn field")
```

Comment:
i dont think so. this error will get hit if the Bearer token has an empty string as the `ucdn` field. a valid Bearer token from our login would only have that if we havent assigned the user a `ucdn`. the `ICDN:UCDN-OVERRIDE` permission is a bypass for that requirement so that people who are admins or have that permission can see the response for any ucdn in order to know what people in those ucdns would get back